frontend/workflows/projectSelector/src/hello-world.tsx

```diff
@@ -84,17 +84,24 @@ const useDispatch = () => {
 
 const selectorReducer = (state: State, action: Action): State => {
   switch (action.type) {
-    case "ADD_PROJECTS":
-      // TODO: don't add if it already exists.
+    case "ADD_PROJECTS": {
+      // a given custom project may already exist in the group so don't trigger a state update for those duplicates
+      const uniqueCustomProjects = action.payload.projects.filter(
+        (project: string) => !(project in state[action.payload.group])
+      );
+      if (uniqueCustomProjects.length === 0) {
+        return { ...state };
```

Comment:
nit: return existing state directly since it was not modified
```suggestion
        return state;
```